### PR TITLE
feat!: use dataframe-like structure for query method output

### LINKED ISSUE #16: Output data structures (Pandas, etc)
Currently, Pandas has to be included as a dependency because it's imported in `dgipy.dgidb`. However, some users might not want to use Pandas at all. This is sort of possible by setting `use_pandas` to False, but you'd still need to have Pandas installed.

It might be more elegant to refactor pandas output (the `_process_output` method) into a separate module that imports pandas in a protected way (i.e. wraps `import pandas` in a try/except block) and then includes pandas as an optional dependency. Could also provide other kinds of data outputs (eg Polars, Dask, or maybe dumping to sqlite or something?) in this way as well.

### LINKED ISSUE #17: Collapse GraphQL output by default
When setting `use_pandas=False`, we return the raw GraphQL output. This includes a bunch of fields that probably* aren't useful and could be collapsed down, like the initial `"data"` property, `"nodes"` etc

*Might be wrong. Gotta think about this

### LINKED ISSUE #18: Add some structure to more complex Pandas columns
Approval ratings get reduced to a long string, but we could probably be storing them as dictionaries within the column

### PR DESCRIPTION
**Pull request by jsstevenson**
close #16 
close #17 
close #18 

Rather than baking Pandas into the main query module, instead all query methods output an easily dataframe-able columnar structure, allowing the user to use their dataframe library of choice instead of locking them into pandas (which has some complexities relating to build package size, versioning issues, etc). 

```pycon
>>> from dgipy import get_gene
>>> result = get_gene(["MYC"])
>>> result
{'name': ['MYC'], 'concept_id': ['hgnc:7553'], 'aliases': [['MYC PROTO-ONCOGENE, BHLH TRANSCRIPTION FACTOR', 'BHLHE39', 'C-MYC', 'MYCC', 'VEGA:OTTHUMG00000128475', 'REFSEQ:NM_001354870', 'ENSEMBL:ENSG00000136997', 'ORPHANET:159785', 'UCSC:UC064QDK.2', 'NCBIGENE:4609', 'CCDS:CCDS6359', 'CCDS:CCDS87627', 'COSMIC:MYC', 'OMIM:190080', 'UNIPROT:P01106', 'MRTL', 'PHARMGKB.GENE:PA31353', 'MYC PROTO-ONCOGENE PROTEIN', 'MYC_HUMAN', 'CIVIC.GID:3737']], 'attributes': [{'MYC AMP': ['Reported Genome Event Targeted'], 'Target Subclass': ['Transcription factor'], 'Target Class': ['Miscellaneous']}]}
>>> import polars as pl
>>> pl.DataFrame(result)
shape: (1, 4)
┌──────┬────────────┬─────────────────────────────────┬─────────────────────────────────┐
│ name ┆ concept_id ┆ aliases                         ┆ attributes                      │
│ ---  ┆ ---        ┆ ---                             ┆ ---                             │
│ str  ┆ str        ┆ list[str]                       ┆ struct[3]                       │
╞══════╪════════════╪═════════════════════════════════╪═════════════════════════════════╡
│ MYC  ┆ hgnc:7553  ┆ ["MYC PROTO-ONCOGENE, BHLH TRA… ┆ {["Reported Genome Event Targe… │
└──────┴────────────┴─────────────────────────────────┴─────────────────────────────────┘
>>> import pandas as pd
>>> pd.DataFrame(result)
  name  ...                                         attributes
0  MYC  ...  {'MYC AMP': ['Reported Genome Event Targeted']...

[1 rows x 4 columns]
```

Sets us up to remove Pandas from main package dependencies, although I think it'll be a requirement for the dashboard stuff (which can be moved to its own package area). 

Removes the option to get the raw GraphQL back. I think that's not very useful, though, I'd prefer to get the structured data as it's given to pandas.

I recognize that this is sort of idiosyncratic without converting to a DF so there's a `type_utils` module with a method that converts this output to a row-oriented version. 

```pycon
>>> from dgipy import get_gene_list
>>> genes = get_gene_list()
>>> from dgipy.type_utils import make_tabular
>>> make_tabular(genes)[0]
{'name': 'MLYCD', 'concept_id': 'hgnc:7150'}
```

This module also includes a quick dump to TSV method.

### CODE REVIEW

**Comment by jsstevenson** [comment]
Oh, a possible TODO for this PR or a new issue is to add a simple CSV dumping method to the `type_utils` module

File: src/dgipy/dgidb.py [hunk 1]
```diff
@@ -29,21 +28,29 @@ def _get_client(api_url: str) -> Client:
...
 
 
+def _group_attributes(row: list[dict]) -> dict:
+    grouped_dict = {}
+    for attr in row:
+        if attr["name"] in grouped_dict:
+            grouped_dict[attr["name"]].append(attr["value"])
+        else:
+            grouped_dict[attr["name"]] = [attr["value"]]
+    return grouped_dict
```
Comment (mcannon068nw) [inline src/dgipy/dgidb.py:38]: testing this part around a little bit. Will this (+the code from get_interactions) basically see all possible keys from a search result and make them as fields within this column for all entries? I noticed right away playing around with the new implementation a lot of nulls everywhere:

Screenshot 2024-08-29 at 8 37 15 AM

I think this is fine and I think I like this implementation better but it makes me wonder: 1) will users be confused if they see nulls (i.e. will they think something broke) and 2) will people be confused if search to search the fields pop in and out (for example what if Cancer Type appears for one search and suddenly the whole column of one particular search result has Cancer Type: None for 95% of entries). These could also not really be concerns but just wanted to think about them. 

Related, all the types of interaction attributes in the db right now:
image

Comment (mcannon068nw) [reply src/dgipy/dgidb.py:38]: example of "pop-in" from get_drug. One of these entries has a "Developer" data point that feels fairly unique, and all other entries have null there now:

Screenshot 2024-08-29 at 8 53 02 AM

**Review by mcannon068nw — APPROVED** [review]
🙌

**Review by mcannon068nw — APPROVED** [review]

**Pull request merged**